packages/host/app/services/card-service.ts

```diff
@@ -164,6 +164,11 @@ export default class CardService extends Service {
     url?: URL,
   ): Promise<SingleCardDocument> {
     let isSaved = !!doc.data.id;
+    if (url && !isSaved) {
+      throw new Error(
+        'Cannot save a card document with no card id. If the card has a URL, it should have an id',
```

Comment:
Should we point out in the error text that the id *is* the URL? Or maybe we can just set it here instead of throwing the error?

Reply:
`url` is not the ID of the document, i believe it is the realm that the card is being saved into. The idea is that for new cards that don't have an ID assigned yet we set this URL to the realm and the realm decides the ID for us. This error seems wrong

Reply:
perhaps we shoudl rename the param to make that more clear